### PR TITLE
test: re run failed pam tests up to 5 times

### PR DESCRIPTION
**Pull request by aadhar-agarwal**
**What type of PR is this?**
/kind test

**What this PR does / why we need it**:
- This PR reruns failed tests in `test_pam.py` up to 5 times
- It utilizes https://github.com/pytest-dev/pytest-rerunfailures
- Added `pytest-rerunfailures` to the requirements.txt file
- This PR is needed to reduce flakiness in `test_pam.py`
  - We need to scrape the console output from running the bash 'login' command
  - An alternative means of testing that does not use console scraping has not been found

**Which issue(s) this PR fixes**:

- This PR helps reduce flakiness in the `Run VHD Tests` step
- An example failure - [Link](https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=98733989&view=logs&j=4d8cf7de-b135-5d7d-7cf0-dcafa581127f&t=f5dce90b-4711-530f-d44e-83a5de1fed6a&s=ff05ad62-bb9a-53b6-ce9f-72f329a63e7c)
- Teams thread in AKS Node Sig - [Link](https://teams.microsoft.com/l/message/19:5ff880aaae474fcba19cbd69b5e9a9ec@thread.skype/1721676174576?tenantId=72f988bf-86f1-41af-91ab-2d7cd011db47&groupId=e121dbfd-0ec1-40ea-8af5-26075f6a731b&parentMessageId=1721676174576&teamName=Azure%20Container%20Compute&channelName=AKS%20Node%20SIG&createdTime=1721676174576)

**How was this change tested?**:
- [Test All VHDs] AKS Linux VHD Build - Msft Tenant runs:
  - https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=98932261&view=results
  - https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=98938578&view=results
  - https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=98945865&view=results

- Created a Mariner AKS cluster, cloned AgentBaker and ran `pytest -v -s --reruns 5 test_pam.py` in a loop 100 times. There was no failure

**Requirements**:

- [x] uses [conventional commit messages](https://www.conventionalcommits.org/)
  
- [ ] includes documentation
- [ ] adds unit tests
- [ ] tested upgrade from previous version

**Special notes for your reviewer**:

**Release note**:

```
none
```

### CODE REVIEW

**Review by corvus-callidus — CHANGES_REQUESTED** [review]

File: vhdbuilder/packer/test/pam/requirements.txt [hunk 1]
```diff
@@ -1,2 +1,3 @@
 pexpect>=4.8.0,<4.9.0
-pytest>=7.3.1,<7.4.0
\ No newline at end of file
+pytest>=7.3.1,<7.4.0
+pytest-rerunfailures>=13.0,<=14.0
```
[the remark this answers is not in the document]
Comment (aadhar-agarwal) [reply vhdbuilder/packer/test/pam/requirements.txt:3]: Talked offline. 14.0 is the most recent tag - https://github.com/pytest-dev/pytest-rerunfailures/tags

We want `>=mostrecentversion,<nextmajorversion.0` -> `>=14.0,<15.0`

[the remark this answers is not in the document]
Comment (aadhar-agarwal) [reply vhdbuilder/packer/test/pam/requirements.txt]: Updated in https://github.com/Azure/AgentBaker/pull/4698/commits/c7a5f7767c02979579c36400e089736ca78592e4

**Comment by r2k1** [comment]
Is there a way to address the test flakiness directly in the test?

I'd like to test fail fast on error. But with this change a failure will take 5x longer to surface.

**Review by anujmaheshwari1 — APPROVED** [review]
lgtm, would be great if you could please address Artur's comment :)

**Comment by aadhar-agarwal** [comment]
> Is there a way to address the test flakiness directly in the test? It just hides the flakiness under the rug.
> 
> I'd like to test fail fast on error. But with this change a failure will take 5x longer to surface.

Yeah, I agree with you. This change is not addressing the actual flakiness in the test. The reason the test is flaky is because we need to interact with the console when logging in as a user or changing the password of a user and the console may be unpredictable. We have not found an alternative approach to using the console. We would need to also understand exactly what extra characters are coming from the console in order to make test_pam.py more robust.

I created a work item on our side to further look into this - https://dev.azure.com/mariner-org/mariner/_workitems/edit/8236

The retry should help make sure the vhd build does not fail out and block a vhd release due to this test.

**Comment by aadhar-agarwal** [comment]
Rebased with master

Re running the pipeline - https://msazure.visualstudio.com/CloudNativeCompute/_build/results?buildId=99061090&view=results

**Review by dallasd1 — APPROVED** [review]

**Review by corvus-callidus — APPROVED** [review]

**Pull request merged**